examples/ml/housing-prices/README.md

```diff
@@ -0,0 +1,247 @@
+
+The **image** defines what Docker image will be used for the pipeline, and the **transform** is the command run once a pipeline job starts.
+
+Once this pipeline is created, it will be looking for any changes to its input, if detected it will start a new job to train given the new dataset.
+
+```bash
+$ pachctl create pipeline -f regression.json
+```
+
+The pipeline will write the output to a PFS repo (`/pfs/out/` in the pipeline json) created with the same name as the pipeline.
+
+### Step 3: Add the housing dataset to the repo
+Now we can add the data which will kick off the processing automatically. If we update the data with a new commit, then the pipeline will automatically re-run. 
+
+```bash
+$ pachctl put file housing_data@master:housing-simplified.csv -f data/housing-simplified-1.csv
+```
+
+We can inspect that the data is in the repository by looking at the files in the repository.
+
+```bash
+$ pachctl list file housing_data@master
+NAME                    TYPE SIZE
+/housing-simplified.csv file 12.14KiB
+```
+
+We can see that the pipeline is running by looking at the status of the job(s). 
+
+```bash
+$ pachctl list job
+ID                               PIPELINE   STARTED        DURATION   RESTART PROGRESS  DL       UL      STATE
+299b4f36535e47e399e7df7fc6ee2f7f regression 23 seconds ago 18 seconds 0       1 + 0 / 1 2.482KiB 1002KiB success
+```
+
+### Step 4: Download files once the pipeline has finished
+Once the pipeline is completed, we can download the files that were created.
+
+```bash
+$ pachctl list file regression@master
+NAME               TYPE SIZE
+/housing-simplified_corr_matrix.png   file 18.66KiB
+/housing-simplified_cv_reg_output.png file 62.19KiB
+/housing-simplified_final_model.sav   file 1.007KiB
+/housing-simplified_pairplot.png      file 207.5KiB
+
+$ pachctl get file regression@master:/ --recursive --output .
+```
+
+When we inspect the learning curve, we can see that there is a large gap between the training score and the validation score. This typically indicates that our model could benefit from the addition of more data. 
+
+<p align="center">
+  <img width="400" height="350" src="images/learning_curve-1.png">
+</p>
+
+So let's update our dataset with additional examples.
+
+### Step 5: Update Dataset
+Here's where pachyderm truely starts to shine. To update our dataset we can run the following command (note that we could also append new examples to the existing file):
+
+```bash
+$ pachctl put file housing_data@master:housing-simplified.csv -f data/housing-simplified-2.csv --overwrite
+```
+
+The new commit of data to the `housing_data` repository automatically kicks off a job on the `regression` pipeline without us having to do anything. 
+
+When the job is complete we can download the new files and see that our model has improved, given the new learning curve. 
```

Comment:
I think we can add a good callout/summary here to data lineage & versioning for experiment tracking. Dont want to make this too much long, but something along the lines of:

Note that because versions all of our input and output data automatically, we can continue to iterate on our data and code and Pachyderm will track all of our experiments. For any given output commit, Pachyderm will tell us exactly which input commit of data was run. In our simple example we only have 2 experiments run so far, but this becomes incredibly important and valuable when we do many more iterations.

Show list commit on the input and output repo. Show inspect commit on the earlier (less good) output commit and experiment and highlight that it includes a reference to the first input commit.

Reply:
I like it. Will add basically exactly what you have here with some examples.

Reply:
Added an additional step to inspect the lineage.